### PR TITLE
lumina.lumina: add code specific to NixOS

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

[Lumina](https://github.com/lumina-desktop/lumina) has a set of [source files targeting specific operating systems](https://github.com/lumina-desktop/lumina/tree/master/src-qt5/core/libLumina). One for NixOS has been submitted at https://github.com/lumina-desktop/lumina/pull/802.

This PR changes de nix expression to use it.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/lumina/lumina/default.nix [hunk 1]
```diff
@@ -34,14 +44,22 @@ mkDerivation rec {
   ];
 
   buildInputs = [
+    acpi
+    alsa-utils
+    brightnessctl
+    coreutils
```
Comment (SuperSandro2000) [inline pkgs/desktops/lumina/lumina/default.nix:50]:
```suggestion
```

This should be part of stdenv already.

File: pkgs/desktops/lumina/lumina/default.nix [hunk 2]
```diff
@@ -53,16 +70,26 @@ mkDerivation rec {
   dontDropIconThemeCache = true;
 
   patches = [
-    ./LuminaOS-NixOS.cpp.patch
+    (fetchpatch {
+      name = "Add-LuminaOS-NixOS.cpp.patch";
+      url = "https://patch-diff.githubusercontent.com/raw/lumina-desktop/lumina/pull/802.patch";
+      sha256 = "sha256-YgTKEvlWd3Ys9IUxcXqvrfV258mXTW/MRhWLj8pP23o=";
```
Comment (SuperSandro2000) [inline pkgs/desktops/lumina/lumina/default.nix:76]:
```suggestion
      sha256 = "0qh9n2gg55axvwih318mmz10b2mgj2cvhm3yj0z8psxszf0cpf5n";
```

File: pkgs/desktops/lumina/lumina/default.nix [hunk 3]
```diff
@@ -53,16 +70,26 @@ mkDerivation rec {
...
-  prePatch = ''
-    # Copy Gentoo setup as NixOS setup and then patch it
-    # TODO: write a complete NixOS setup?
-    cp -a src-qt5/core/libLumina/LuminaOS-Gentoo.cpp src-qt5/core/libLumina/LuminaOS-NixOS.cpp
-  '';
-
   postPatch = ''
+    # Fix file paths in NixOS distribution specific source code
+    acpi="${acpi}/bin/acpi" \
+      amixer="${alsa-utils}/bin/amixer" \
```
Comment (SuperSandro2000) [inline pkgs/desktops/lumina/lumina/default.nix:83]: I think it would be easier if we used substituteAll in patchPhase with inherits.

**Comment by romildo** [comment]
Closing for now.

**Pull request closed without merging**